class-gf-razorpay.php

```diff
@@ -0,0 +1,285 @@
+    {
+        if (self::$_instance == null)
+        {
+            self::$_instance = new GFRazorpay();
+        }
+
+        return self::$_instance;
+    }
+
+    public function init_frontend()
+    {
+        parent::init_frontend();
+
+        add_action('gform_after_submission', array($this, 'pay_using_razorpay'), 10, 2);
+    }
+
+    public function plugin_settings_fields()
+    {
+        return array(
+            array(
+                'title'       => '',
+                'fields'      => array(
+                    array(
+                        'name'    => self::GF_RAZORPAY_KEY,
+                        'label'   => esc_html__( 'Razorpay Key', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'name'    => self::GF_RAZORPAY_SECRET,
+                        'label'   => esc_html__( 'Razorpay Secret', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'type' => 'save',
+                        'messages' => array(
+                            'success' => esc_html__( 'Settings have been updated.', 'razorpay-gravity-forms' )
+                        ),
+                    ),
+                ),
+            ),
+        );
+    }
+
+    public function get_customer_fields_array($feed, $entry)
+    {
+        $fields = array();
+
+        foreach ($this->get_customer_fields() as $field)
+        {
+            $field_id = $feed['meta'][$field['meta_name']];
+
+            $value = rgar($entry, $field_id);
+
+            $fields[$field['meta_name']] = $value;
+        }
+
+        return $fields;
+    }
+
+    public function get_customer_fields()
+    {
+        return array(
+            array('name' => 'first_name', 'label' => 'First Name', 'meta_name' => 'billingInformation_firstName'),
+            array('name' => 'last_name', 'label' => 'Last Name', 'meta_name' => 'billingInformation_lastName'),
+            array('name' => 'email', 'label' => 'Email', 'meta_name' => 'billingInformation_email'),
+            array('name' => 'address1', 'label' => 'Address', 'meta_name' => 'billingInformation_address'),
+            array('name' => 'address2', 'label' => 'Address 2', 'meta_name' => 'billingInformation_address2'),
+            array('name' => 'city', 'label' => 'City', 'meta_name' => 'billingInformation_city'),
+            array('name' => 'state', 'label' => 'State', 'meta_name' => 'billingInformation_state'),
+            array('name' => 'zip', 'label' => 'Zip', 'meta_name' => 'billingInformation_zip'),
+            array('name' => 'country', 'label' => 'Country', 'meta_name' => 'billingInformation_country'),
+        );
+    }
+
+    public function callback()
+    {
+        $entry_id = $_COOKIE['entry_id'];
```

Comment:
Please change cookie key names to something that uses `razorpay_gf` or something similar as prefix. Very high chance of someone else attempting to use `entry_id`. And since this seems to be used across multiple places, maybe convert to a constant as well?